overlord/overlord.go

```diff
@@ -92,21 +92,33 @@ type Overlord struct {
 	proxyConf func(req *http.Request) (*url.URL, error)
 }
 
+// RestartBehavior controls how to hanndle and carry forward restart requests
+// via the state.
+type RestartBehavior interface {
+	HandleRestart(t state.RestartType)
+	// RebootVerified is called early with information whether a reboot
+	// was requested but snapd restarted without it happening.
+	// It is always called with the state lock held.
+	RebootVerified(st *state.State, expectedRebootDidNotHappen bool) error
```

Comment:
I wonder if it would make sense to split this and have one "ExpectedRebootDidNotHappen" and one "RebootDone" or something like this? Something like  http://paste.ubuntu.com/p/CjzzSpdhNn/ which I find slightly easier to read. Its slightly bigger though (two functions on the interface instead of just one for the boot validation). Anyway, just a suggestion.

Reply:
I did something like that but a bit differently. One thing is that we call the positive case also if there was no pending reboot at all, so I had to fix the naming accordingly.